src/library/tabledelegates/keydelegate.cpp

```diff
@@ -0,0 +1,61 @@
+#include "library/tabledelegates/keydelegate.h"
+
+#include <qnamespace.h>
+
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "moc_keydelegate.cpp"
+#include "track/keyutils.h"
+#include "util/color/predefinedcolorpalettes.h"
+#include "util/color/rgbcolor.h"
+
+KeyDelegate::KeyDelegate(QTableView* pTableView)
+        : TableItemDelegate(pTableView) {
+}
+
+void KeyDelegate::paintItem(
+        QPainter* painter,
+        const QStyleOptionViewItem& option,
+        const QModelIndex& index) const {
+    paintItemBackground(painter, option, index);
+
+    QString keyText = index.data().toString();
```

Comment:
Here you should request the color directly from the model using a role, the get around the guessing below.

Reply:
yes, guessing the key from the text is inefficient and slow.

Reply:
Ah, okay. So just get the raw value (key code) via index->data(), cast to int and get the color from `KeyUtils::colorFromNumericValue(int)`?

Reply:
= don't get the **color** from the model, as requested, just the value

Reply:
yeah, though I wouldn't even go over the `int` index and just store and retrieve the `mixxx::track::io::key::ChromaticKey` directly. otherwise we'll have to deal with out-of-bound values and so forth again.